# U.S. Medical Insurance Costs

This is a data analysis with python using the csv module to extract some data for the insurance cost of some people, Trying to identify some factors and make a answer to  why is that result happening

In [77]:
import csv
age= []
children = []
smoker = []
sex = []
location = []
with open('insurance.csv', 'r') as insurance_data:
    insurance = csv.DictReader(insurance_data)   
    for data in insurance:
        age.append(data['age'])
        children.append(data['children'])
        smoker.append(data['smoker'])
        sex.append(data['sex'])
        location.append(data['region'])
    
    

In [39]:
#lets write a function and a variable to store and reopen the csv file to loop through it
def get_file():
    with open('insurance.csv', 'r') as insurance_data:
        insurance = csv.DictReader(insurance_data) 

  Now that we have the data we can meake our analysis and make our questions to answer

In [86]:
#lets take the average age of the data
count = 0
for i in age:
    count += int(i)
average = count/len(age)
print("The average age in the data is: ",average)

The average age in the data is:  39.20702541106129


# So now we now the average age that is 39 years old, now lets find out the most common region

In [64]:
#lets make our variables wo we can see all the regions in the dataset without looking at the dataframe
regiones = {}
for region in location:
  if region not in regiones:
    regiones[region] = 1
  else:
    regiones[region] += 1

print(regiones)        

{'southwest': 325, 'southeast': 364, 'northwest': 325, 'northeast': 324}


 This show us that southeast is the most common region in the data so now lets get the average charges in this region

In [85]:
charges = 0
charg = []
with open('insurance.csv', 'r') as insurance_data:
    insurance = csv.DictReader(insurance_data)   
    for data in insurance:
        if data["region"] == "southeast":
            charges += float((data['charges']))
            charg.append(data['charges'])
        
    print("The average price in the most common region is: ", charges/len(charg))

The average price in the most common region is:  14735.411437609895


 lets find out how much is the difference if you have more than 1 kid  so we can get a estimation for a person that has 1, or more than 1.

In [116]:
promedio2 = 0
prom2 = []
promedio1 = 0
prom1 = []
with open('insurance.csv', 'r') as insurance_data:
    insurance = csv.DictReader(insurance_data)
    for i in insurance:
        if int(i['children']) >= 2:
            prom2.append(i['charges'])
            promedio2 = promedio2 + float(i['charges'])
        if int(i['children']) <= 1:
            prom1.append(i['charges'])
            promedio1 = promedio1 + float(i['charges'])
            
    print("The average  charges when you have 2 children or more is : ",promedio2/len(prom2))
    print("The avergae charges when you have 1 or dont have kids is: ",promedio1/len(prom1))
        
        
            
            

The average  charges when you have 2 children or more is :  14847.39845902273
The avergae charges when you have 1 or dont have kids is:  12497.739052103547


# As you can see an estimate of  the difference of the  price if you have more than one kid is around 2200 dollars. now lets see how much afects to be a smoker or non smoker

In [31]:

def smokers_in_total():
    is_smoker = 0
    non_smoker  = 0
    for i in smoker:
       if i == "yes":
        is_smoker += 1
        
       else:
        non_smoker += 1
    print("In the data there are: ",is_smoker, " smokers in total and ", non_smoker, " non smokers")

In [32]:
smokers_in_total()

In the data there are:  274  smokers in total and  1064  non smokers


In the results we can see a huge amount of people that doesn't smoke in comparison to people that smoke, from this point, there are two main questions i want to analyse. 

- How much does smoking increases the price of your insurance

- The older the age, is it more probable that you smoke?


In [75]:
def get_smoking_price():
    
    with open('insurance.csv', 'r') as insurance_data:
        insurance = csv.DictReader(insurance_data)  
        smoker_charges = 0
        non_smoker_charges = 0
        for data in insurance:
            if data['smoker'] == "yes":
                smoker_charges += float(data['charges'])
            if data['smoker'] == "no":
                non_smoker_charges  += float(data['charges'])
    print(smoker_charges, " ",non_smoker_charges)
            
    

In [76]:
get_smoking_price()

8781763.52184   8974061.468918996


# As you can see the price is almost the same, but we need to remember that there ar only 274 smokers on the list, and 1064 non smokers so that says that smoking could make your charges increase almost by 4 times than a non smoker person.

In [86]:
def age_and_smoking():
     with open('insurance.csv', 'r') as insurance_data:
        insurance = csv.DictReader(insurance_data)
        smokers_age = 0
        smokers_len = 0
        non_smokers_age = 0
        non_smokers_len = 0
        for data in insurance:
            if data['smoker'] == "yes":
                smokers_age += int(data['age'])
                smokers_len += 1
            if data['smoker'] == "no":
                non_smokers_age += int(data['age'])
                non_smokers_len += 1
        print("The average age of the people that smoke in the data is: ",smokers_age/smokers_len, " and the average age of the non smokers is ", non_smokers_age/non_smokers_len)
        
        #now lets check the average age of non smokers
        
            
    

In [87]:
age_and_smoking()

The average age of the people that smoke in the data is:  38.51459854014598  and the average age of the non smokers is  39.38533834586466


so the average age in both cases is almost the same, for now we cant tell if the age is a factor that leads to smoking